datadog/resource_datadog_service_level_objective.go

```diff
@@ -149,25 +149,71 @@ func ValidateServiceLevelObjectiveTypeString(v interface{}, k string) (ws []stri
 
 func buildServiceLevelObjectiveStruct(d *schema.ResourceData) *datadog.ServiceLevelObjective {
 
-	slo := datadog.ServiceLevelObjective{
-		Type: datadog.String(d.Get("type").(string)),
-		Name: datadog.String(d.Get("name").(string)),
+	slo := &datadog.ServiceLevelObjective{
+		ID: datadog.String(d.Id()),
+	}
+
+	if attr, ok := d.GetOk("name"); ok {
+		slo.SetName(attr.(string))
```

Comment:
Minor nit: `name` is required, so we don't have to check for its presence, we can just `slo.SetName(d.Get("name"))`